packages/chakra-components/src/ComboBox/index.js

```diff
@@ -1,130 +1,372 @@
-import React, { useRef } from "react";
-import useSelect from "./useSelect";
-import styled from "@emotion/styled";
+import React, { forwardRef, cloneElement, useRef, createContext } from "react";
+import {
+  Box,
+  PseudoBox,
+  Icon,
+  Input,
+  InputRightElement,
+  InputLeftElement,
+  InputLeftAddon,
+  InputRightAddon,
+  useTheme,
+} from "@chakra-ui/core";
+import Popper from "@chakra-ui/core/dist/Popper";
 import matchSorter from "match-sorter";
 import { FixedSizeList } from "react-window";
+import { useForkRef, cleanChildren } from "@chakra-ui/core/dist/utils";
+import { useComboBoxContext } from "./useComboBoxContext";
+import useSelect from "./useSelect";
+
+import { inputSizes } from "@chakra-ui/core/dist/Input/styles";
+import { useComboBoxPopperStyle, useComboBoxOptionStyle } from "./styles";
+
+/* =========================================================================
+  ComboBoxContext
+  ========================================================================== */
+
+export const ComboBoxContext = createContext();
+
+/* =========================================================================
+  ComboBox
+  ========================================================================== */
+
+const ComboBox = forwardRef(
+  (
+    {
+      value,
+      options,
+      onChange,
+      pageSize = 10,
+      itemHeight = 40,
+      children,
+      size = "md",
+      ...props
+    },
+    ref,
+  ) => {
+    const { sizes } = useTheme();
+    const height = inputSizes[size] && inputSizes[size]["height"];
+    let pl = null;
+    let pr = null;
+    let right = null;
+    const validChildren = cleanChildren(children);
+
+    const reactWindowInstanceRef = useRef(null);
+    const optionsRef = useRef(null);
+
+    const scrollToIndex = index => {
+      if (!reactWindowInstanceRef.current) {
+        return;
+      }
+      reactWindowInstanceRef.current.scrollToItem(index);
+    };
```

Comment:
@navin-moorthy  To change the scroll order to top, this was missing in your code

```suggestion
    const scrollToIndex = (index, position) => {
      if (!reactWindowInstanceRef.current) {
        return;
      }
      reactWindowInstanceRef.current.scrollToItem(index, position);
    };
```

Reply:
Yup, it works. Thanks.